debug/Debug.js

```diff
@@ -0,0 +1,130 @@
+function Debug(scene) {
+    // CHARTS
+    // create charts div
+    const chartDiv = document.createElement('div');
+    chartDiv.id = 'chart-div';
+    chartDiv.style = 'position: absolute; bottom: 0; left: 0; width: 100vw; height: 20rem; background-color: white';
+
+    scene.viewerDiv.appendChild(chartDiv);
+
+    // line graph for nb elements
+    const nbObjectsCanvas = document.createElement('canvas');
+    nbObjectsCanvas.heigth = '20rem';
+    nbObjectsCanvas.width = '40rem';
+    nbObjectsCanvas.id = 'nb-objects';
+    chartDiv.appendChild(nbObjectsCanvas);
+
+    const timestamp = Date.now();
+    const nbObjectsDataset = { label: 'Number of object in Scene', data: [{ x: 0, y: 0 }] };
+    const nbObjectsChartLabel = ['0s'];
+    const nbObjectsChart = new Chart('nb-objects', {
+        type: 'line',
+        data: {
+            labels: nbObjectsChartLabel,
+            datasets: [nbObjectsDataset],
+        },
+        options: {
+            animation: { duration: 10 },
+            scales: {
+                yAxes: [{
+                    display: true,
+                    ticks: {
+                        suggestedMin: 0, // minimum will be 0, unless there is a lower value.
+                    },
+                }],
+            },
+        },
+    });
+
+    function updateNbObjectsChart() {
+        function countElem(node) {
+            if (!node) {
+                return 0;
+            }
+            let count = 1; // this node
+            if (node.children) {
+                for (const child of node.children) {
+                    count += countElem(child);
+                }
+            }
+            return count;
+        }
+
+        const timeInS = Math.floor((Date.now() - timestamp) / 1000);
+
+        const limit = 50;
+        nbObjectsChartLabel.push(`${timeInS}s`);
+        if (nbObjectsChartLabel.length > limit) {
+            nbObjectsChartLabel.shift();
+        }
+        nbObjectsDataset.data.push({ x: timeInS, y: countElem(scene.gfxEngine.scene3D) });
+        if (nbObjectsDataset.data.length > limit) {
+            nbObjectsDataset.data.shift();
+        }
+        nbObjectsChart.update();
+    }
+
+    // hook that to scene.update
+    const oldUpdate = Object.getPrototypeOf(scene).update.bind(scene);
+    scene.update = function debugUpdate() {
+        oldUpdate();
+        updateNbObjectsChart();
+    };
+
+    // DEBUG CONTROLS
+    const gui = new dat.GUI();
+
+    const state = {
+        showOutline: false,
+        wireframe: false,
+        displayCharts: true,
+    };
+
+    // charts
+    gui.add(state, 'displayCharts').name('Display charts').onChange((newValue) => {
+        if (newValue) {
+            chartDiv.style.display = 'flex';
+        } else {
+            chartDiv.style.display = 'none';
+        }
+    });
+
+    // tiles outline
+    gui.add(state, 'showOutline').name('Show tiles outline').onChange((newValue) => {
+        var cO = function c0(object) {
```

Comment:
To elaborate on why I put that in the prototype: the current default value of wireframe and showOutline change at runtime (when we tick the boxes) and is to be shared by all instances of LayeredMaterial. Having them at the layer creation is not enough (and imho, the layerconfiguration should not depend on debug stuff). For these reasons, the prototype is a really good place for these booleans. If you're concerned about the fact we do it multiple times, I agree it's not necessary. I can make a test that will prevent any further assignment after the first one if you want, but I think we don't really care (setting boolean might even be less costly than making a test).